### PR TITLE
SEQNG-782 Avoid processing empty streams.

### PR DESCRIPTION
**Pull request by jluhrs**
Every processed event can produce new events, in the form of an`Option[Stream]`. The code was queuing a `Stream` even if the result was `None`. I removed that.
There is also some code formatting.

### CODE REVIEW

File: modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala [hunk 1]
```diff
@@ -263,38 +263,45 @@ class Engine[D, U](stateL: Lens[D, Engine.State]) {
...
+  def mapEvalState[A, S, B](input: Stream[IO, A],
+                            initialState: S, f: (A, S) => IO[(S, B, Option[Stream[IO, A]])])
+                           (implicit ec: ExecutionContext): Stream[IO, B] = {
     Stream.eval(fs2.async.unboundedQueue[IO, Stream[IO, A]]).flatMap { q =>
       Stream.eval_(q.enqueue1(input)) ++
         q.dequeue.joinUnbounded.evalMapAccumulate(initialState) { (s, a) =>
           f(a, s).flatMap {
-            case (ns, b, st) =>
-              q.enqueue1(st) >> IO.pure((ns, b))
+            case (ns, b, None)     => IO.pure((ns, b))
```
Comment (cquiroz) [inline modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala]: is this the core change?

File: modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala [hunk 2]
```diff
@@ -263,38 +263,45 @@ class Engine[D, U](stateL: Lens[D, Engine.State]) {
...
+            case (ns, b, Some(st)) => q.enqueue1(st) >> IO.pure((ns, b))
```
Comment (cquiroz) [inline modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala]: It is a bit strange that you could change from `q.enqueue1(Option[A])` to `q.enqueue1(A)`. Am I missing something?

Comment (jluhrs) [reply modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala]: Yes, you are missing something. Previously, `st` was a `Stream` that most of the time was `Stream.empty`. Now it is an `Option[Stream]`, where the case when it was `Stream.empty` was replaced by `None`.

File: modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala (hunk 1 above)
Comment (jluhrs) [reply modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala]: This, in conjunction with the change in line 298.

**Pull request merged**